### PR TITLE
Add a vertical swipe refresh to lists of reviews

### PR DESCRIPTION
**Pull request by jmettler27**
This fixes the issue where the user could not see its review appear on the list of reviews and had to exit and re-enter the Activity
Furthermore the user can refresh manually the list by swiping down on the screen.

### CODE REVIEW

File: app/src/main/java/com/github/sdp/ratemyepfl/viewmodel/RoomReviewsListViewModel.kt [hunk 1]
```diff
@@ -37,4 +37,8 @@ class RoomReviewsListViewModel @Inject constructor(
     fun getReviews(): LiveData<List<Review?>> {
         return reviewsLiveData
     }
+
+    fun refreshReviewsList() {
```
Comment (mbouilloud) [inline app/src/main/java/com/github/sdp/ratemyepfl/viewmodel/RoomReviewsListViewModel.kt:41]: Why make updateReviewList private and then use this function to call it ?

Comment (AlexandreMessmer) [reply app/src/main/java/com/github/sdp/ratemyepfl/viewmodel/RoomReviewsListViewModel.kt:41]: I agree, this adds some boilerplate ^^

File: app/src/main/java/com/github/sdp/ratemyepfl/activity/classrooms/RoomReviewsListActivity.kt [hunk 2]
```diff
@@ -60,6 +63,14 @@ class RoomReviewsListActivity : AppCompatActivity() {
                 { fab.shrink() }
             )
         )
+
+        // Vertical swipe refreshes the list of reviews
+        swipeRefresher = findViewById(R.id.swiperefresh)
+        swipeRefresher.setOnRefreshListener {
+            viewModel.refreshReviewsList()
+            swipeRefresher.isRefreshing = false
```
Comment (AlexandreMessmer) [inline app/src/main/java/com/github/sdp/ratemyepfl/activity/classrooms/RoomReviewsListActivity.kt:71]: I don't really get why you need to set isRefreshing to false. I think I'll have a look at the documentation.

**Review by AlexandreMessmer — APPROVED** [review]
Goob job in general. I think you spent way more time on how to implement this swipe rather than implementing it. There is still a problem of duplication because of those ReviewList, and I think we may switch to a common, global, and generic ReviewList(Fragment) where the logic of swiping is directly implemented. I only have one comment. I think we should write a test for this swipeRefresher. I know that Espresso does not provide ViewActions for swiping (or I guess at least). On the main branch, I created a CustomViewActions object where we can provide ViewActions for our tests.

**Pull request merged**